recipes/_agent_common_directories.rb

```diff
@@ -12,8 +17,11 @@
     end
     recursive true
     if node['platform_family'] == 'windows'
-      notifies :run, 'execute[install_zabbix_agentd]'
-      notifies :run, 'execute[start_zabbix_agentd]'
+      if services.count < 1
+        notifies :run, 'execute[install_zabbix_agentd]'
```

Comment:
This is a 4-ish level nested conditional. That is unnecessarily complex. First, we should have an outer-level conditional for the platform family. It's usually nicest to have a `case` statement for that.

For example, see what happens when we just move the `platform_family` check outside the `directory` resource:

```ruby
  case node['platform_family']
  when 'linux'
    directory dir do
      owner 'root'
      group 'root'
      mode '755'
      recursive true
      notifies :restart, 'service[zabbix_agentd]'
    end
  when 'windows'
    # windows stuff
  end
```

The `linux` case requires no conditionals at all! That's fantastic - and easy to read. We should strive for that.

Now, in the `windows` case, I really hope we have a way to get rid of the `if services.count < 1` line. I think the way you do that is remove the `action :nothing` from `execute[install_zabbix_agent]` and instead use a `not_if` block:

```ruby
require 'win32ole'
zabbixservice = 'Zabbix Agent'
wmi = WIN32OLE.connect("winmgmts://")

execute 'install_zabbix_agentd' do
  command "#{node['zabbix']['agent']['agentd_dir']} --config \"#{node['zabbix']['agent']['config_file']}\" --install"
  not_if  { wmi.ExecQuery("Select * from Win32_Service where Name = '#{zabbixservice}'").count == 0 }
end
```

There's no reason Zabbix installation has to be triggered by a notification. In fact, that adds more execution paths for our program that aren't necessary ... if we're running this recipe, we know that we want to install the Zabbix agent. So make sure you're including the recipe that installs the agent, and check there if the agent is  installed.

No need to add `notifies` to every block anywhere to reinstall the agent, unless we _actually_ need to reinstall every time we change a config file. Is that the case?